libs/s25main/ingameWindows/iwStatistics.cpp

```diff
@@ -62,34 +63,10 @@ iwStatistics::iwStatistics(const GameWorldViewer& gwv)
         if(!curPlayer.isUsed())
             continue;
 
-        switch(curPlayer.nation)
-        {
-            case Nation::Africans:
-                AddImageButton(1 + i, DrawPoint(startX + pos * 34 - 17, 45 - 23), Extent(34, 47), TextureColor::Green1,
-                               LOADER.GetImageN("io", 257), curPlayer.name)
-                  ->SetBorder(false);
-                break;
-            case Nation::Japanese:
-                AddImageButton(1 + i, DrawPoint(startX + pos * 34 - 17, 45 - 23), Extent(34, 47), TextureColor::Green1,
-                               LOADER.GetImageN("io", 253), curPlayer.name)
-                  ->SetBorder(false);
-                break;
-            case Nation::Romans:
-                AddImageButton(1 + i, DrawPoint(startX + pos * 34 - 17, 45 - 23), Extent(34, 47), TextureColor::Green1,
-                               LOADER.GetImageN("io", 252), curPlayer.name)
-                  ->SetBorder(false);
-                break;
-            case Nation::Vikings:
-                AddImageButton(1 + i, DrawPoint(startX + pos * 34 - 17, 45 - 23), Extent(34, 47), TextureColor::Green1,
-                               LOADER.GetImageN("io", 256), curPlayer.name)
-                  ->SetBorder(false);
-                break;
-            case Nation::Babylonians:
-                AddImageButton(1 + i, DrawPoint(startX + pos * 34 - 17, 45 - 23), Extent(34, 47), TextureColor::Green1,
-                               LOADER.GetImageN("io_new", 7), curPlayer.name)
-                  ->SetBorder(false);
-                break;
-        }
+        const auto& portrait = Portraits[curPlayer.portraitIndex];
```

Comment:
this index *must* be checked if its in range. (We're not on C++26 where the behavior *may* be defined)

Reply:
yes. everywhere where it can be set is best practice for this kind of usage (game engine)

Reply:
I can't seem to find it. Can you link to it?

Reply:
Wrong diff version, sorry